app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -0,0 +1,61 @@
+module CourseBasicDetailConcern
+  extend ActiveSupport::Concern
+
+  included do
+    decorates_assigned :course
+    before_action :build_provider, only: :new
+    before_action :build_course, only: %i[edit update]
+  end
+
+  def new
+    # Using .find(:new) here is a little bit hacky, but this is the only way I
+    # could find to construct the URL with `.../courses/new` at the end, and at
+    # the end of the day jsonapi defines ids as being strings so it's in no way
+    # invalid. If we can find a way to use custom enpoints or other to improve
+    # this we should.
+    @course = Course
```

Comment:
There might be some value in refactoring the `Course.where...` out into a method that explains what it's doing a bit more clearly imo

Reply:
Ya, I think you're right, as evidenced by the explanatory note here. I've moved it to `Course.fetch_new`.